server/src/main/resources/spring/login-ui.xml

```diff
@@ -189,7 +182,13 @@
                 <ref bean="uaaAuthenticationFailureHandler"/>
                 <bean class="org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler"/>
                 <bean class="org.springframework.security.web.csrf.CsrfLogoutHandler">
-                    <constructor-arg ref="loginCookieCsrfRepository"/>
+                    <constructor-arg>
+                        <bean class="org.springframework.security.web.csrf.LazyCsrfTokenRepository">
+                            <constructor-arg>
+                                <bean class="org.springframework.security.web.csrf.HttpSessionCsrfTokenRepository"/>
```

Comment:
Before this PR, we had a bean of `CookieBasedCsrfTokenRepository` which was auto-wired into this handler. You replaced that bean with an **_inline_** `HttpSessionCsrfTokenRepository`. Can you just change the class of that auto-wired bean instead?

So replace this bean:
```xml
<bean id="loginCookieCsrfRepository"
      class="org.cloudfoundry.identity.uaa.security.web.CookieBasedCsrfTokenRepository">
    <property name="secure" value="${require_https:false}"/>
</bean>
```
with this one (adding the lazy wrapper, if required):
```xml
<bean id="httpSessionCsrfTokenRepository"
      class="org.springframework.security.web.csrf.HttpSessionCsrfTokenRepository">
</bean>
```
---
The advantage would be that you can keep using the mock repository in **unit tests**. No intrusive changes will be needed (like adding GETs).

You will just need to replace this mock:
```java
@Bean
CookieBasedCsrfTokenRepository loginCookieCsrfRepository() {
    return mock(CookieBasedCsrfTokenRepository.class);
}
```
with this one:
```java
@Bean
HttpSessionCsrfTokenRepository httpSessionCsrfTokenRepository() {
    return mock(HttpSessionCsrfTokenRepository.class);
}
```

Reply:
There is in fact only one unit test that has changed - InvitationsControllerTest, the one which defines the bean you mentioned above, and one unit test that has been deleted.
All the other tests that have changed are mockMvc and integration tests.
These tests test the the server using real components, where feasible, to give us the confidence that the real server is behaving correctly.